pkg/ebpf/alu_instructions.go

```diff
@@ -39,37 +39,39 @@ func newAluInstruction[T Src](oc pb.AluOperationCode, insclass pb.InsClass, dst
 		   https://docs.kernel.org/bpf/standardization/instruction-set.html#bit-immediate-instructions
 		*/
 	case int64:
-		upper := int32(src >> 32)
-		return &pb.Instruction{
-			Opcode: &pb.Instruction_MemOpcode{
-				MemOpcode: &pb.MemOpcode{
-					Mode:             pb.StLdMode_StLdModeIMM,
-					Size:             pb.StLdSize_StLdSizeDW,
-					InstructionClass: pb.InsClass_InsClassLd,
+		if oc == pb.AluOperationCode_AluMov {
+			upper := int32(src >> 32)
+			return &pb.Instruction{
+				Opcode: &pb.Instruction_MemOpcode{
+					MemOpcode: &pb.MemOpcode{
+						Mode:             pb.StLdMode_StLdModeIMM,
+						Size:             pb.StLdSize_StLdSizeDW,
+						InstructionClass: pb.InsClass_InsClassLd,
+					},
 				},
-			},
-			DstReg:    dst,
-			SrcReg:    pb.Reg_R0,
-			Offset:    0,
-			Immediate: int32(src),
-			PseudoInstruction: &pb.Instruction_PseudoValue{
-				PseudoValue: &pb.Instruction{
-					Opcode: &pb.Instruction_MemOpcode{
-						MemOpcode: &pb.MemOpcode{
-							Mode:             0,
-							Size:             0,
-							InstructionClass: 0,
+				DstReg:    dst,
+				SrcReg:    pb.Reg_R0,
+				Offset:    0,
+				Immediate: int32(src),
+				PseudoInstruction: &pb.Instruction_PseudoValue{
+					PseudoValue: &pb.Instruction{
+						Opcode: &pb.Instruction_MemOpcode{
+							MemOpcode: &pb.MemOpcode{
+								Mode:             0,
+								Size:             0,
+								InstructionClass: 0,
+							},
+						},
+						DstReg:    0,
+						SrcReg:    0,
+						Offset:    0,
+						Immediate: upper,
+						PseudoInstruction: &pb.Instruction_Empty{
+							Empty: &pb.Empty{},
 						},
-					},
-					DstReg:    0,
-					SrcReg:    0,
-					Offset:    0,
-					Immediate: upper,
-					PseudoInstruction: &pb.Instruction_Empty{
-						Empty: &pb.Empty{},
 					},
 				},
-			},
+			}
```

Comment:
should we put an else clause here where you set the value of imm to the int32 cast of src?

Reply:
So if an instruction is generated (that is not Mov64) with a int64 immediate it gets treated as a int32 immediate?